pkg/action/scan_error.go

```diff
@@ -0,0 +1,80 @@
+package action
+
+import "fmt"
+
+// Error message constants for NewFileReportError reasons.
+// If the compiled rules are invalid or the scanner malfunctions, yrs.Scan will fail.
+// If mrs is nil or Generate fails, the file report will be nil.
+const (
+	errMsgUnknown        = "unknown error"
+	errMsgScanFailed     = "scan failed"
+	errMsgGenerateFailed = "failed to generate file report"
+)
+
+type ErrorType int
+
+// Error type iotas.
+// TypeUnknown will be the default of `0`.
+// TypeScanError is to be used when compiled rules are invalid or the scan fails otherwise.
+// TypeGenerateError is to be used when a file's report cannot be created.
+const (
+	TypeUnknown ErrorType = iota
+	TypeScanError
+	TypeGenerateError
+)
```

Comment:
```suggestion
const (
	// TypeUnknown will be the default of `0`.
	TypeUnknown ErrorType = iota
	// TypeScanError is to be used when compiled rules are invalid or the scan fails otherwise.
	TypeScanError
	// TypeGenerateError is to be used when a file's report cannot be created.
	TypeGenerateError
)
```

Needs to be inlined for godoc to know how to map them to each type.

Reply:
Excellent -- thanks!